client/src/main/scala/caliban/client/GraphQLRequest.scala

```diff
@@ -10,12 +10,6 @@ case class GraphQLRequest(query: String, variables: Map[String, __Value])
 
 object GraphQLRequest {
 
-  implicit val encoder: Encoder[GraphQLRequest] = (req: GraphQLRequest) =>
-    Json.obj(
-      "query"     -> Json.fromString(req.query),
-      "variables" -> Json.obj(req.variables.map { case (k, v) =>
-        k -> v.asJson
-      }.toList: _*)
-    )
+  implicit val jsonEncoder: JsonValueCodec[GraphQLRequest] = JsonCodecMaker.makeCirceLike[GraphQLRequest]
```

Comment:
Do we need this to be `CirceLike` (and the other Codecs) or is it only to keep the encoding / decoding as similar as what we previously had?

Reply:
This is just to keep the codecs as close to the original behavior. I'm not sure if there would be a noticeable difference but I feel it would be hard to decode if these changed in subtle ways under the hood.